### ISSUE TITLE
[New rule] `updateRenderObject` should be consistent with `createRenderObject`

### ISSUE THREAD

**Issue by fzyzcjy** [dart-code-checker/dart-code-metrics#996]
**Please describe what the rule should do:**

I have commonly seen (and I also sometimes did it...) that, the updateRenderObject forget to update some fields that should be updated. This is subtle to be found, because as long as the createRenderObject works, nobody will realize the bug, until one day that field is updated and updateRenderObject does not do the job. So time will be wasted later to debug and find this.

**If your rule is inspired by other please provide link to it:**

**What category of rule is this? (place an "X" next to just one item)**

- [x] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

case 1:

```dart
class MyWidget extends SingleChildRenderObjectWidget {
  final A a;
  final B b;

  const MyWidget({ ... });

  @override
  MyRender createRenderObject(BuildContext context) {
    return MyRender(a: a, b: b);
  }

  @override
  void updateRenderObject(BuildContext context, RenderOnLayout renderObject) {
    renderObject.a = a;
    // LINT: you forget to update the `b` field!
  }
}
```

case 2 (less common though):

```dart
class MyWidget extends SingleChildRenderObjectWidget {
  final A a;
  final B b;

  const MyWidget({ ... });

  @override
  MyRender createRenderObject(BuildContext context) {
    return MyRender(a: a, b: b);
  }

  // LINT: you forget the whole updateRenderObject
}
```

**Are you willing to submit a pull request to implement this rule?**

**Comment by fzyzcjy** [comment]
A million. 

https://api.flutter.dev/flutter/rendering/RenderBox-class.html has "Implementers":
[PlatformViewRenderBox](https://api.flutter.dev/flutter/rendering/PlatformViewRenderBox-class.html)[RenderCustomMultiChildLayoutBox](https://api.flutter.dev/flutter/rendering/RenderCustomMultiChildLayoutBox-class.html)[RenderEditable](https://api.flutter.dev/flutter/rendering/RenderEditable-class.html)[RenderErrorBox](https://api.flutter.dev/flutter/rendering/RenderErrorBox-class.html)[RenderFlex](https://api.flutter.dev/flutter/rendering/RenderFlex-class.html)[RenderFlow](https://api.flutter.dev/flutter/rendering/RenderFlow-class.html)[RenderImage](https://api.flutter.dev/flutter/rendering/RenderImage-class.html)[RenderListBody](https://api.flutter.dev/flutter/rendering/RenderListBody-class.html)[RenderListWheelViewport](https://api.flutter.dev/flutter/rendering/RenderListWheelViewport-class.html)[RenderParagraph](https://api.flutter.dev/flutter/rendering/RenderParagraph-class.html)[RenderPerformanceOverlay](https://api.flutter.dev/flutter/rendering/RenderPerformanceOverlay-class.html)[RenderProxyBox](https://api.flutter.dev/flutter/rendering/RenderProxyBox-class.html)[RenderRotatedBox](https://api.flutter.dev/flutter/rendering/RenderRotatedBox-class.html)[RenderShiftedBox](https://api.flutter.dev/flutter/rendering/RenderShiftedBox-class.html)[RenderStack](https://api.flutter.dev/flutter/rendering/RenderStack-class.html)[RenderTable](https://api.flutter.dev/flutter/rendering/RenderTable-class.html)[RenderUiKitView](https://api.flutter.dev/flutter/rendering/RenderUiKitView-class.html)[RenderViewportBase](https://api.flutter.dev/flutter/rendering/RenderViewportBase-class.html)[RenderWrap](https://api.flutter.dev/flutter/rendering/RenderWrap-class.html)[TextureBox](https://api.flutter.dev/flutter/rendering/TextureBox-class.html)

**Comment by fzyzcjy** [comment]
```dart
class Padding extends SingleChildRenderObjectWidget {
  /// Creates a widget that insets its child.
  ///
  /// The [padding] argument must not be null.
  const Padding({
    super.key,
    required this.padding,
    super.child,
  }) : assert(padding != null);

  /// The amount of space by which to inset the child.
  final EdgeInsetsGeometry padding;

  @override
  RenderPadding createRenderObject(BuildContext context) {
    return RenderPadding(
      padding: padding,
      textDirection: Directionality.maybeOf(context),
    );
  }

  @override
  void updateRenderObject(BuildContext context, RenderPadding renderObject) {
    renderObject
      ..padding = padding
      ..textDirection = Directionality.maybeOf(context);
  }

  @override
  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
    super.debugFillProperties(properties);
    properties.add(DiagnosticsProperty<EdgeInsetsGeometry>('padding', padding));
  }
}

**Comment by fzyzcjy** [comment]
A ton of widgets in flutter. For example, the Opacity:

```dart
class Opacity extends SingleChildRenderObjectWidget {
  /// Creates a widget that makes its child partially transparent.
  ///
  /// The [opacity] argument must not be null and must be between 0.0 and 1.0
  /// (inclusive).
  const Opacity({
    super.key,
    required this.opacity,
    this.alwaysIncludeSemantics = false,
    super.child,
  }) : assert(opacity != null && opacity >= 0.0 && opacity <= 1.0),
       assert(alwaysIncludeSemantics != null);

  /// The fraction to scale the child's alpha value.
  ///
  /// An opacity of 1.0 is fully opaque. An opacity of 0.0 is fully transparent
  /// (i.e., invisible).
  ///
  /// The opacity must not be null.
  ///
  /// Values 1.0 and 0.0 are painted with a fast path. Other values
  /// require painting the child into an intermediate buffer, which is
  /// expensive.
  final double opacity;

  /// Whether the semantic information of the children is always included.
  ///
  /// Defaults to false.
  ///
  /// When true, regardless of the opacity settings the child semantic
  /// information is exposed as if the widget were fully visible. This is
  /// useful in cases where labels may be hidden during animations that
  /// would otherwise contribute relevant semantics.
  final bool alwaysIncludeSemantics;

  @override
  RenderOpacity createRenderObject(BuildContext context) {
    return RenderOpacity(
      opacity: opacity,
      alwaysIncludeSemantics: alwaysIncludeSemantics,
    );
  }

  @override
  void updateRenderObject(BuildContext context, RenderOpacity renderObject) {
    renderObject
      ..opacity = opacity
      ..alwaysIncludeSemantics = alwaysIncludeSemantics;
  }

  @override
  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
    super.debugFillProperties(properties);
    properties.add(DoubleProperty('opacity', opacity));
    properties.add(FlagProperty('alwaysIncludeSemantics', value: alwaysIncludeSemantics, ifTrue: 'alwaysIncludeSemantics'));
  }
}

**Comment by incendial** [comment]
Available in 4.19.0 🚀

**Comment by fzyzcjy** [comment]
🚀
